pkg/daemon/ceph/osd/daemon.go

```diff
@@ -349,12 +349,24 @@ func getAvailableDevices(context *clusterd.Context, desiredDevices []DesiredDevi
 					// the desired devices is a regular expression
 					matched, err = regexp.Match(desiredDevice.Name, []byte(device.Name))
 					if err != nil {
-						logger.Errorf("regex failed on device %s and filter %s. %+v", device.Name, desiredDevice.Name, err)
+						logger.Errorf("regex failed on device %q and filter %q. %+v", device.Name, desiredDevice.Name, err)
 						continue
 					}
-					logger.Infof("device %s matches device filter %s: %t", device.Name, desiredDevice.Name, matched)
+					logger.Debugf("device %q matches device filter %q: %t", device.Name, desiredDevice.Name, matched)
+				} else if desiredDevice.IsDevlinksFilter {
+					for _, devlink := range strings.Fields(device.DevLinks) {
+						matched, err = regexp.Match(desiredDevice.Name, []byte(devlink))
+						if err != nil {
+							logger.Errorf("regex failed on device %q and filter %q. %+v", device.Name, desiredDevice.Name, err)
+							continue
+						}
+						if matched {
+							break
```

Comment:
As commented below, I will log the final result of filtering in the following part of the code.
It would be useful to log the alias and the filter for debugging here, but the current logging
style of "outputting log even if not matched" is somewhat noisy for the inner loop.
So I will record device aliases in one line after this loop.